qlty-check/src/parser/reek.rs

```diff
@@ -0,0 +1,98 @@
+use super::Parser;
+use anyhow::Result;
+use qlty_types::analysis::v1::{Category, Issue, Level, Location, Range};
+use serde::{Deserialize, Serialize};
+
+pub struct Reek {}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ReekJson {
+  pub context: String,
+  pub lines: Vec<i32>,
+  pub message: String,
+  pub smell_type: String,
+  pub source: String,
+  pub documentation_link: String,
+}
```

Comment:
Is there documentation on this JSON format available?

Reply:
None that I found, but there is this spec:

https://github.com/troessner/reek/blob/master/spec/reek/report/json_report_spec.rb

Reply:
I'd add a code comment to the parser linking to that

A main thing to watch out for is not accidentally depending on a field always being present when it is present only in some conditions.

But I also wouldn't recommend using defensive coding on every field as that is cumbersome and really reduces the information density of the parser